### PR TITLE
Correctly handle AXX style errors and send to client

### PR DESCRIPTION
**Pull request by adelowo**
This PR allows us correctly catch AXX style errors like A14 and others. Including the ability to correctly send the error
back to the client immediately without enforcing them to wait for long

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and hash computation for user operations, improving reliability and performance.
- **Bug Fixes**
	- Improved handling of errors during gas estimation and transaction reverts.
- **Refactor**
	- Streamlined the logic flow in the `SendUserOperation` function for better readability and maintainability.
	- Transitioned to a more standardized error handling approach across multiple components.

### CODE REVIEW

File: srv/relayer.go [hunk 1]
```diff
@@ -120,15 +122,39 @@ func (r *Relayer) SendUserOperation() modules.BatchHandlerFunc {
...
+					}
+
+					return OpHashes{
+						Error:  mergedErr,
+						Solved: currentOpHash,
+						Trx:    hash,
+					}, false
+				})
+			}
+		}
```
Comment (blewater) [inline srv/relayer.go]: I like agree with coderabbit's comment :) I found it hard to follow which variable is named vs captured and  will improve the readability by using named parameters (ctx.Batch vs ops) and avoid 2 level closures.

File: srv/relayer.go [hunk 2]
```diff
@@ -120,15 +122,39 @@ func (r *Relayer) SendUserOperation() modules.BatchHandlerFunc {
 			attribute.Int64("wait_timeout", int64(r.waitTimeout)),
 		)
 
+		var computeOPHashes = func(err error, hash common.Hash) {
```
Comment (blewater) [inline srv/relayer.go]: Please add a func documentation and a few comments for the context and what we're accomplishing here.

File: srv/relayer.go [hunk 3]
```diff
@@ -120,15 +122,39 @@ func (r *Relayer) SendUserOperation() modules.BatchHandlerFunc {
...
+
+			for _, op := range ctx.Batch {
+
+				currentOpHash, unsolvedOpHash := utils.GetUserOpHash(op, r.ep, r.chainID)
+
+				r.m.Compute(unsolvedOpHash, func(oldValue OpHashes, loaded bool) (newValue OpHashes, delete bool) {
+
+					var mergedErr error
+					if err != nil {
+						mergedErr = multierror.Append(oldValue.Error, err)
```
Comment (blewater) [inline srv/relayer.go]: optional you can eliminate a dep by using [errors join](https://pkg.go.dev/errors#Join)

**Review by blewater — CHANGES_REQUESTED** [review]
Good, a bit of improvement on readability

**Review by hpsing — APPROVED** [review]

**Review by blewater — APPROVED** [review]
Good!

**Pull request merged**